modules/rostests/apitests/shell32/LocaleTests.cpp

```diff
@@ -0,0 +1,290 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     LGPL-2.1+ (https://spdx.org/licenses/LGPL-2.1+)
+ * PURPOSE:     Formal locale verification tests
+ * COPYRIGHT:   Copyright 2024 Stanislav Motylkov <x86corez@gmail.com>
+ */
+
+#include "shelltest.h"
+
+#include <winnls.h>
+#include <strsafe.h>
+
+#include <set>
+#include <map>
+
+enum E_MODULE
+{
+    shell32,
+    userenv,
+    syssetup,
+    mmsys,
+    explorer_old,
+};
+
+enum E_STRING
+{
+    SH32_PROGRAMS,
+    SH32_STARTUP,
+    SH32_STARTMENU,
+    SH32_PROGRAM_FILES,
+    SH32_PROGRAM_FILES_COMMON,
+    SH32_ADMINTOOLS,
+    UENV_STARTMENU,
+    UENV_PROGRAMS,
+    UENV_STARTUP,
+    SYSS_PROGRAMFILES,
+    SYSS_COMMONFILES,
+    MMSY_STARTMENU,
+    EOLD_PROGRAMS,
+};
+
+typedef struct PART_TEST
+{
+    E_MODULE eModule;
+    UINT id;
+    SIZE_T nParts;
+} PART_TEST;
+
+typedef struct PART
+{
+    E_STRING Num;
+    UINT Idx;
+} PART;
+
+typedef struct PART_MATCH
+{
+    PART p1, p2;
+} PART_MATCH;
+
+LCID curLcid = 0;
+std::set<LANGID> langs;
+std::map<E_MODULE, HMODULE> mod;
+std::map<E_STRING, PART_TEST> parts =
+{
+    { SH32_PROGRAMS, { shell32, 45 /* IDS_PROGRAMS "Start Menu\Programs" */, 2 } },
+    { SH32_STARTUP, { shell32, 48 /* IDS_STARTUP "Start Menu\Programs\StartUp" */, 3 } },
+    { SH32_STARTMENU, { shell32, 51 /* IDS_STARTMENU "Start Menu" */, 1 } },
+    { SH32_PROGRAM_FILES, { shell32, 63 /* IDS_PROGRAM_FILES "Program Files" */, 1 } },
+    { SH32_PROGRAM_FILES_COMMON, { shell32, 65 /* IDS_PROGRAM_FILES_COMMON "Program Files\Common Files" */, 2 } },
+    { SH32_ADMINTOOLS, { shell32, 67 /* IDS_ADMINTOOLS "Start Menu\Programs\Administrative Tools" */, 3 } },
+    { UENV_STARTMENU, { userenv, 11 /* IDS_STARTMENU "Start Menu" */, 1 } },
+    { UENV_PROGRAMS, { userenv, 12 /* IDS_PROGRAMS "Start Menu\Programs" */, 2 } },
+    { UENV_STARTUP, { userenv, 13 /* IDS_STARTUP "Start Menu\Programs\StartUp" */, 3 } },
+    { SYSS_PROGRAMFILES, { syssetup, 3600 /* IDS_PROGRAMFILES "%SystemDrive%\Program Files" */, 2 } },
+    { SYSS_COMMONFILES, { syssetup, 3601 /* IDS_COMMONFILES "Common Files" */, 1 } },
+    { MMSY_STARTMENU, { mmsys, 5851 /* IDS_STARTMENU "Start Menu" */, 1 } },
+    { EOLD_PROGRAMS, { explorer_old, 10 /* IDS_PROGRAMS "Programs" */, 1 } },
+};
```

Comment:
error C2440: 'initializing': cannot convert from 'initializer list' to 'std::map<E_STRING,PART_TEST,std::less<E_STRING>,std::allocator<std::pair<const E_STRING,PART_TEST>>>'

Reply:
[parts.h.zip](https://github.com/user-attachments/files/16173865/parts.h.zip)
Use this. You can manually add them.

Reply:
I'm not sure such defines would make the code better. A person in another chat suggested to try tweaking MSVC compiler flags.